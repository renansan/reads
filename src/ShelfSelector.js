import React from 'react'
import PropTypes from 'prop-types'
import shelfsList from './shelfs'

/**
 * ShelfSelector
 * A select to change the book's shelf
 *
 * This component must receive a function to handle select changes and the
 * current shelf of the book. The list of shelfs are taken from the shelfs.json.
 *
 * @param {Object} props
 */
const ShelfSelector = (props) => {
  //handle changes on ShelfSelector
  const handleChange = (event) => {
    // Set el from event target and shelf ID and title in an object
    const el = event.target;
    const shelf = {
      id: el.value,
      title: el[el.selectedIndex].textContent
    }
    props.handleChange(shelf, event);
  }
  return (
    <select onChange={handleChange} value={props.currentShelf || 'none'}>
      <option value="move" disabled>Move to...</option>
      {shelfsList.shelfs && shelfsList.shelfs.map((bookshelf, index) => (
        <option key={index} value={bookshelf.id}>{bookshelf.title}</option>
      ))}
      <option value="none">None</option>
    </select>
  )
}

ShelfSelector.propTypes = {
  handleChange: PropTypes.func.isRequired,
  currentShelf: PropTypes.string.isRequired,
}

export default ShelfSelector
