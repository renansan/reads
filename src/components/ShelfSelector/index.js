import React from 'react'
import PropTypes from 'prop-types'
import shelfsList from './shelfs'

/**
 * ShelfSelector
 * A select to change the book's shelf
 *
 * This component must receive a function to handle select changes and the
 * current shelf of the book. The list of shelfs are taken from the shelfs.json.
 *
 * @param {Object} props
 */
const ShelfSelector = (props) => {
  return (
    <select onChange={props.handleChange} value={props.currentShelf || 'none'}>
      <option value="move" disabled>Move to...</option>
      {shelfsList.shelfs && shelfsList.shelfs.map((bookshelf, index) => (
        <option key={index} value={bookshelf.id}>{bookshelf.title}</option>
      ))}
      <option value="none">None</option>
    </select>
  )
}

ShelfSelector.propTypes = {
  handleChange: PropTypes.func.isRequired,
  currentShelf: PropTypes.string.isRequired,
}

export default ShelfSelector
