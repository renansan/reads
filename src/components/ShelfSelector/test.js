import React from 'react'
import renderer from 'react-test-renderer';
import ShelfSelector from './index'

it('renders correctly', () => {
  const props = {
    handleChange: jest.fn(() => 'handleChange'),
    currentShelf: 'test',
  }
  const tree = renderer.create(<ShelfSelector {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
