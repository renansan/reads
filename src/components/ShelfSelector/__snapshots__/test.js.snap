// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<select
  onChange={[Function]}
  value="test"
>
  <option
    disabled={true}
    value="move"
  >
    Move to...
  </option>
  <option
    value="currentlyReading"
  >
    Currently Reading
  </option>
  <option
    value="wantToRead"
  >
    Want to Read
  </option>
  <option
    value="read"
  >
    Read
  </option>
  <option
    value="none"
  >
    None
  </option>
</select>
`;
